identity/src/main/java/com/android/identity/credentialtype/CredentialTypeRepository.kt

```diff
@@ -28,18 +28,55 @@ package com.android.identity.credentialtype
  */
 class CredentialTypeRepository {
     private val credentialTypes: MutableList<CredentialType> = mutableListOf()
+    private val mdocDataElements: MutableMap<String, Map<String, Map<String, MdocDataElement>>> =
+        mutableMapOf()
 
     /**
      * Add a Credential Type to the repository.
      */
-    fun addCredentialType(credentialType: CredentialType){
+    fun addCredentialType(credentialType: CredentialType) {
         credentialTypes.add(credentialType)
+        if (credentialType.mdocCredentialType != null) {
+            mdocDataElements[credentialType.mdocCredentialType.docType] =
+                credentialType.mdocCredentialType.namespaces.map {
+                    Pair(
+                        it.namespace,
+                        it.dataElements.associateBy { el -> el.attribute.identifier }.toMap()
+                    )
+                }.toMap()
+        }
```

Comment:
So we're keeping an extra set of data elements in this class, in addition to each type? This seems wrong. 

But more, importantly, I don't think `getMdocDataElement()` should be on the repository in the first place. Why not put it on `MdocCredentialType` and then add a lookup function for mDL doctype so the code becomes:
```
val mdocCredentialType = repo.getMdocCredentialType(docType)
val mdocDataElement = mdocCredentialType?.getMdoc(namespaceName, dataElementName)
```

Reply:
I changed the Lists to Map in the context of the MdocCredentialType. In the context of the overall CredentialType the usage of a Map is arbitrary. The key could be the mdoc doctype , but also a VC Credential Type, so I kept a here a List and added a method to lookup the VC Credential Type.

Reply:
By changing the Lists to Maps, a separate method getMdocElements is not needed anymore